### ISSUE TITLE
Do not force value to be set on Variable when there is a VariableReference

### ISSUE THREAD

**Issue by apupier** [microsoft/debug-adapter-protocol#242]
When there is a VariableReference, it means that it is expected to have children. These children are usually providing a good representation. And a better one than a "multiline string".

for reference VS Code is not using this value anyway and it is causing troubles when not setting it https://github.com/microsoft/vscode/issues/141544

**Comment by weinand** [comment]
@apupier Why do you think that VS Code ignores the `value` property on structured variables? This is not true. Here is an example from the JavaScript debugger:

![CleanShot 2022-01-26 at 14 28 02@2x](https://user-images.githubusercontent.com/1898161/151171562-65474367-519a-43c2-ad99-daee0324a55d.png)

**Comment by apupier** [comment]
> Why do you think that VS Code ignores the `value` property on structured variables? This is not true

my bad. Seems I have not tested correctly what I wanted.

I still think that it will ease creation of Debug Adapter servers if the value is not mandatory in case of Variable Reference is set.

**Comment by weinand** [comment]
@apupier I agree that making the `value` attribute optional would simplify the implementation of a debug adapter that has nothing to show for structured objects. However, in my experience those debug adapters are rare because it is always good practice to show some identifying name for structured objects, e.g. some object ID, address (or just a one-line representation consisting of the first three property values). 

If we want to make the `value` attribute optional in a non-breaking way, we would have to introduce a new client-capability `supportsOptionalVariableValue`. With this a DAP client (e.g. VS Code) can announce that it can handle a missing `value` attribute. Without that announcement, debug adapters must continue to provide the attribute in order to be backward compatible.

But this need for backward compatibility would defeat the reason we wanted to introduce the change in the first place: to make development easier.

So I suggest we don't change the spec but I will clarify the description of the `value` attribute by mentioning the empty string when no vlaue needs to be shown.

/cc @connor4312 @roblourens

**Comment by apupier** [comment]
> So I suggest we don't change the spec but I will clarify the description of the `value` attribute by mentioning the empty string when no valaue needs to be shown.

sounds a great compromise, especially with the drawbacks you mentioned.

**Comment by weinand** [comment]
Here is the new description for the `value` property of `Variable`:
```
/** The variable's value.
    This can be a multi-line text, e.g. for a function the body of a function.
    For structured variables (which do not have a simple value), it is recommended to provide 
    a one line representation of the structured object. This helps to identify the structured
    object in the collapsed state when its children are not yet visible.
    An empty string can be used if no value should be shown in the UI.
*/
value: string;
```
